server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/AddProjectAction.java

```diff
@@ -61,9 +63,13 @@ public void define(WebService.NewController controller) {
   @Override
   public void handle(Request request, Response response) throws Exception {
     AddProjectRequest addProjectRequest = toWsRequest(request);
-    String profileKey = projectAssociationFinder.getProfileKey(addProjectRequest.getLanguage(), addProjectRequest.getProfileName(), addProjectRequest.getProfileKey());
-    String projectUuid = projectAssociationFinder.getProjectUuid(addProjectRequest.getProjectKey(), addProjectRequest.getProjectUuid());
-    profileProjectOperations.addProject(profileKey, projectUuid, userSession);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      String profileKey = projectAssociationFinder.getProfileKey(addProjectRequest.getLanguage(), addProjectRequest.getProfileName(), addProjectRequest.getProfileKey());
```

Comment:
I agree, but it's not loaded twice when parameter `profileKey` is set. For this reason, as you suggested, I prefer handle this refactoring later.